fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java

```diff
@@ -596,4 +650,506 @@ public static Expression divideDecimalV3(DecimalV3Literal first, DecimalV3Litera
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.round(0).getDouble()));
+    }
+
+    /**
+     * round
+     */
+    @ExecFunction(name = "round")
+    public static Expression round(DoubleLiteral first, IntegerLiteral second) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.round(second.getValue()).getDouble()));
+    }
+
+    /**
+     * ceil
+     */
+    @ExecFunction(name = "ceil")
+    public static Expression ceil(DecimalV3Literal first) {
+        return first.roundCeiling(0);
+    }
+
+    /**
+     * ceil
+     */
+    @ExecFunction(name = "ceil")
+    public static Expression ceil(DecimalV3Literal first, IntegerLiteral second) {
+        return first.roundCeiling(second.getValue());
+    }
+
+    /**
+     * ceil
+     */
+    @ExecFunction(name = "ceil")
+    public static Expression ceil(DoubleLiteral first) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.roundCeiling(0).getDouble()));
+    }
+
+    /**
+     * ceil
+     */
+    @ExecFunction(name = "ceil")
+    public static Expression ceil(DoubleLiteral first, IntegerLiteral second) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.roundCeiling(second.getValue()).getDouble()));
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DecimalV3Literal first) {
+        return first.roundFloor(0);
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DecimalV3Literal first, IntegerLiteral second) {
+        return first.roundFloor(second.getValue());
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DoubleLiteral first) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.roundFloor(0).getDouble()));
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DoubleLiteral first, IntegerLiteral second) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return filterNanResult(new DoubleLiteral(middleResult.roundFloor(second.getValue()).getDouble()));
+    }
+
+    /**
+     * exp
+     */
+    @ExecFunction(name = "exp")
+    public static Expression exp(DoubleLiteral first) {
+        return filterNanResult(new DoubleLiteral(Math.exp(first.getValue())));
```

Comment:
only nan input will lead to nan result? so why filter nan here?

Reply:
input check was done